src_app.py

```diff
@@ -1,41 +1,139 @@
+
+    # create the parser and set the usage and the description
+    parser = argparse.ArgumentParser(description=description, usage=usage)
+
+    # define an argument to be passed to parse and get it
+    parser.add_argument('-path', dest='model_path', type=str, required=True,
+                        help='the pickle file path')
+
+    # get the argument which is the model file path
+    args = parser.parse_args()
+
+    # return the model file path
+    return args.model_path
+
+
+def read_linear_regression_model(pickle_file_path):
+    """read the model form the path passed in arguments
+
+    Args:
+        pickle_file_path (string): the pickle file path
+
+    Returns:
+        sklearn.linear_model._base.LinearRegression: the linear regression model
+    """
+    # read the linear regression model from the pickle file
+    linear_regression_model = pickle.load(open(pickle_file_path, 'rb'))
+
+    # return the model
+    return linear_regression_model
+
+
+def predict(linear_regression_model, house_features):
+    """predinc the house price using the linear regression model for the house
+       features
+
+    Args:
+        linear_regression_model ( sklearn.linear_model._base.LinearRegression):
+            the linear regression model
+        house_features (dict): the house features to predict for
+
+    Returns:
+        dict: the predict value stored in dict
+    """
+
+    # slice the features values
+    features_values = [float(house_features.get(
+        feature)) for feature in config.FEATURES_NAMES_IN_ORDER]
+
+    # create a numpy array of the features values and reshape it into 2D,
+    # due to he predict function assumes a 2D array where each line is a
+    # sample to predict for
+    features_values_in_shape = np.array(features_values).reshape(1, 6)
+
+    # make the prediction
+    predicted_house_price = linear_regression_model.predict(
+        features_values_in_shape)
+
+    # slice the predicted value
+    json_reslut = {config.PREDICTED_VALUE: predicted_house_price[0]}
+
+    # return the predicted value
+    return json_reslut
+
+
 @app.route('/')
-def hello():
+def welcoming():
     return render_template('index.html')
 
 
 @app.route('/predict', methods=['GET', 'POST'])
 def predict_house_price():
 
+    # if the method is POST
     if request.method == 'POST':
 
-        # import the predicting module
-
-        import linear_model
-
         # get the data from the POST request
-
         house_features = request.get_json()
 
         # call the predicting function and return the result
+        return json.dumps(predict(linear_regression_model, house_features))
 
-        return linear_model.predict(house_features, parameters)
+    # if the method is GET
     else:
 
-        return render_template('index.html')
+        # get the house_features from the get request
+        house_features = {
+            feature: request.args.get(feature)
+            for feature in config.FEATURES_NAMES_IN_ORDER
```

Comment:
You can do this without using any static variables from config => " for argument in request.args.keys()"